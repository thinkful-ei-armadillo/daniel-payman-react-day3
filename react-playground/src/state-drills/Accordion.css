ul {
  list-style: none;
}

li .card-header {
  padding: 0.75rem 1.25rem;
  margin-bottom: 0;
  background-color: rgba(0, 0, 0, 0.1);
  border-bottom: 1px solid rgba(0, 0, 0, 0.125);
}

li {
  background-color: #fff;
  background-clip: border-box;
  border: 1px solid rgba(0, 0, 0, 0.125);
  border-radius: 0.25rem;
}

li p {
  padding: 10px;
}

li button {
  font-weight: 800;
  color: #007bff;
  height: 50px;
  border: 1px solid transparent;
  background-color: rgba(0, 0, 0, 0);
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  line-height: 1.5;
}
