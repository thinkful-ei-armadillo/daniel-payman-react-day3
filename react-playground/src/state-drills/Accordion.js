import React from 'react';
import './Accordion.css';

class Accordion extends React.Component {
  state = {
    currentIndex: null
  };

  static defaultProps = {
    sections: []
  };

  generateSectionElements = () => {
    if (this.props.sections.length === 0) {
      return <li />;
    }
    const sections = this.props.sections;

    return sections.map((element, index) => {
      let ariaExpanded = false;
      if (this.state.currentIndex === index) {
        ariaExpanded = true;
      }
      return (
        <li key={index}>
          <div className="card-header">
            <button
              aria-expanded={ariaExpanded}
              onClick={() => this.setSectionIndex(index)}
            >
              {element.title}
            </button>
          </div>
          {this.state.currentIndex === index && <p>{element.content}</p>}
        </li>
      );
    });
  };

  setSectionIndex = index => {
    if (index === this.state.currentIndex) {
      index = null;
    }
    this.setState({
      currentIndex: index
    });
  };

  render() {
    return <ul>{this.generateSectionElements()}</ul>;
  }
}

export default Accordion;
