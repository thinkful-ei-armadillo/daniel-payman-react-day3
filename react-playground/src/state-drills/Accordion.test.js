import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Accordion from './Accordion';
import state from '../state';

describe('Accordion component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordion />, div);
    ReactDOM.unmountComponentAtNode(div);
  });


//   The component renders an empty li when the sections prop is not supplied
  it('renders the no-prop UI as expected', () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

// The component renders no sections as active by default
  it('renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={state.sections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })

// The component opens a clicked section
  it('opens section on click', () => {
    const wrapper = shallow(<Accordion sections={state.sections} />);
    wrapper.find('button').at(0).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot();
  })

// The component only opens the last section when multiple sections have been clicked.
it('opens section on click', () => {
  const wrapper = shallow(<Accordion sections={state.sections} />);
  wrapper.find('button').at(0).simulate('click')
  wrapper.find('button').at(1).simulate('click')
  expect(toJson(wrapper)).toMatchSnapshot();
})

});