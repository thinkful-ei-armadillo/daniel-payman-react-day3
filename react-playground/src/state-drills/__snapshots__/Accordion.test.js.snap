// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component opens section on click 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component opens section on click 2`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders no sections as active by default 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Accordion component renders the no-prop UI as expected 1`] = `
<ul>
  <li />
</ul>
`;
